openfe/protocols/openmm_rfe/_rfe_utils/relative.py

```diff
@@ -229,6 +232,164 @@ def __init__(self,
+
+        logger.info("CMAPTorsionForce found checking compatibility")
+        # add an assert to check the types to make mypy happy
+        assert cmap_old is not None and cmap_new is not None
+        # some quick checks on compatibility like the number of maps and total number of terms
+        old_num_maps = cmap_old.getNumMaps()
+        new_num_maps = cmap_new.getNumMaps()
+        if old_num_maps != new_num_maps:
+            raise RuntimeError(
+                f"Incompatible CMAPTorsionForce between end states expected to have same number of maps, "
+                f"found old: {old_num_maps} and new: {new_num_maps}")
+
+        old_num_torsions = cmap_old.getNumTorsions()
+        new_num_torsions = cmap_new.getNumTorsions()
+        if old_num_torsions != new_num_torsions:
+            raise RuntimeError(
+                f"Incompatible CMAPTorsionForce between end states expected to have same number of torsions, "
+                f"found old: {old_num_torsions} and new: {new_num_torsions}")
+
+        return cmap_old, cmap_new, old_num_maps, new_num_maps, old_num_torsions, new_num_torsions
+
+    def _handle_cmap_torsion_force(self):
+        """
+        This method does the following in order:
+            - Some basic checks that the CMAPTorsionForce exists in both old/new systems.
+            - Adds the CMAPTorsionForce from the old system
+            - Checks that the new system CMAPTorsionForce terms are equal to the old system's (we do not allow for alchemically changing CMAP terms).
+        """
+        cmap_old = self._old_system_forces.get("CMAPTorsionForce", None)
+        cmap_new = self._new_system_forces.get("CMAPTorsionForce", None)
+
+        if cmap_new == cmap_old is None:
+            logger.info("No CMAPTorsionForce found skipping adding force.")
+            return
+
+        # verify compatibility and extract numbers of maps and torsions
+        (
+            cmap_old,
+            cmap_new,
+            old_num_maps,
+            new_num_maps,
+            old_num_torsions,
+            new_num_torsions
+        ) = self._verify_cmap_compatibility(
+            cmap_old, cmap_new
+        )
+
+        logger.info("Adding CMAPTorsionForce to hybrid system")
+        # start looping through the old system terms and add them to the hybrid system
+        # track the terms we add so we can cross compare with the new system and also make sure we don't hit
+        # an index in the alchemical region
+        hybrid_cmap_force = openmm.CMAPTorsionForce()
+        self._hybrid_system.addForce(hybrid_cmap_force)
+        self._hybrid_system_forces["cmap_torsion_force"] = hybrid_cmap_force
+
+        old_system_maps = {}
+        old_system_terms = {}
+        # gather all alchemical atoms, use a copy so we don't change the groups
+        alchemical_atoms = self._atom_classes["core_atoms"].copy()
+        alchemical_atoms.update(self._atom_classes["unique_old_atoms"], self._atom_classes["unique_new_atoms"])
```

Comment:
[nit] since this is mostly for checking, can you move it closer to line 362?

Reply:
Good idea!